tests/ui.rs

```diff
@@ -98,8 +114,89 @@ fn miri_config(target: &str, path: &str, mode: Mode, with_dependencies: bool) ->
         ..Config::rustc(path)
     };
 
+    if let Mode::Pass { rustc: true } = mode {
+        config.comment_defaults.base().add_custom("run", Run {
+            exit_code: 0,
+            output_conflict_handling: Some(|path, actual, errors, config| {
+                let path = path.with_file_name(
+                    path.file_name().unwrap().to_str().unwrap().replace(".run.", "."),
+                );
+                // Blessing is only allowed in miri mode, rustc mode must match
+                ui_test::error_on_output_conflict(&path, actual, errors, config);
+            }),
+        });
+        config.program.envs.push(("MIRI_BE_RUSTC".into(), Some("host".into())));
+
+        #[derive(Debug)]
+        struct Strip;
+
+        impl Flag for Strip {
+            fn clone_inner(&self) -> Box<dyn Flag> {
+                Box::new(Strip)
+            }
+
+            fn must_be_unique(&self) -> bool {
+                true
+            }
+
+            fn apply(
+                &self,
+                cmd: &mut Command,
+                _config: &TestConfig,
+                _build_manager: &BuildManager,
+            ) -> Result<(), Errored> {
+                let mut c = Command::new(cmd.get_program());
+                for (k, v) in cmd.get_envs() {
+                    match v {
+                        Some(v) => c.env(k, v),
+                        None => c.env_remove(k),
+                    };
+                }
+                if let Some(dir) = cmd.get_current_dir() {
+                    c.current_dir(dir);
+                }
+                c.args(
+                    cmd.get_args().filter(|arg| !arg.as_encoded_bytes().starts_with(b"-Zmiri-")),
+                );
+                *cmd = c;
+                Ok(())
+            }
+
+            fn test_condition(
+                &self,
+                _config: &Config,
+                comments: &Comments,
+                revision: &str,
+            ) -> bool {
+                for rev in comments.for_revision(revision) {
+                    for arg in &rev.compile_flags {
```

Comment:
Shouldn't we just strip everything that starts with `-Zmiri`?

Reply:
This is filtering out tests. Otherwise I need to add `only-miri` a lot

Reply:
Not sure what that changes. I would still have to do all the same changes in ui.rs

I will do that to reduce the scope of this PR and future unrelated PRs that may be affected, but the impl is just as complex

Reply:
Well the logic for implicitly making tests `only-miri` when they use these particular flag could be removed, I think?